src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs

```
@@ -193,6 +198,9 @@ public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, string>> conf
             {
                 throw new ArgumentException($"Invalid value '{basicAuthSource}' specified for property '{SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource}'");
             }
+            if (username != null && password != null && authenticator == null)
```

coding style is to always use `{ }`'s

---

you should do more checking - throw an argument exception on all invalid input cases.